src/main/java/org/opensearch/security/user/UserService.java

```diff
@@ -178,4 +202,79 @@ private String generatePassword() {
         String generatedPassword = "superSecurePassword";
         return generatedPassword;
     }
+
+    /**
+     * This function retrieves the auth token associated with a service account.
+     * Fails if the provided account is not a service account or account is not enabled.
+     *
+     * @param accountName A string representing the name of the account
+     * @return A string auth token
+     */
+    public String generateAuthToken(String accountName) throws IOException {
+
+        final SecurityDynamicConfiguration<?> internalUsersConfiguration = load(getUserConfigName(), false);
+
+        if (!internalUsersConfiguration.exists(accountName)) {
+            throw new UserServiceException(FAILED_ACCOUNT_RETRIEVAL_MESSAGE);
+        }
+
+        String authToken = null;
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            JsonNode accountDetails = mapper.readTree(internalUsersConfiguration.getCEntry(accountName).toString());
+            final ObjectNode contentAsNode = (ObjectNode) accountDetails;
```

Comment:
While it isn't required to an ObjectNode, JsonNode can use `withObject` to create a new object with the modifications [1]; however what you are doing here is fine.

- [1] https://fasterxml.github.io/jackson-databind/javadoc/2.14/com/fasterxml/jackson/databind/JsonNode.html#withObject-com.fasterxml.jackson.core.JsonPointer-

Reply:
Oh good to know, I will take a look into whether it makes more sense to swap over or not. This was the previous pattern before the refactor but that does not mean it is the "correct" pattern.